packages/react-reconciler/src/ReactFiberBeginWork.old.js

```diff
@@ -2754,13 +2754,15 @@ function updateDehydratedSuspenseComponent(
   }
 }
 
-function scheduleWorkOnFiber(fiber: Fiber, renderLanes: Lanes) {
+function scheduleSuspenseWorkOnFiber(fiber: Fiber, renderLanes: Lanes) {
   fiber.lanes = mergeLanes(fiber.lanes, renderLanes);
   const alternate = fiber.alternate;
   if (alternate !== null) {
     alternate.lanes = mergeLanes(alternate.lanes, renderLanes);
   }
-  scheduleWorkOnParentPath(fiber.return, renderLanes);
+  // Guaranteed to be non-empty since the fiber is not a root.
+  const parentFiber: Fiber = (fiber.return: any);
+  scheduleContextWorkOnParentPath(parentFiber, renderLanes, parentFiber);
```

Comment:
This doesn't make sense to me. If the propagation root is the same as the fiber you're scheduling work on, wouldn't it always immediately bail out?

The propagation root should is SuspenseList in this case. I would pass it in as an argument from the caller.

Though I'm confused why the tests wouldn't catch that, so maybe I'm missing something.

Reply:
Tests might not have caught it because it only matters if there's a bailout (sCU, pure) in between the SuspenseList and the child so it's likely we don't have coverage for this combination of events

Reply:
>If the propagation root is the same as the fiber you're scheduling work on, wouldn't it always immediately bail out?

Why would it? It's a while loop and the propagation root check is at the end so it would do one iteration for the current node and then bail. Is that not what we want? Maybe I don't understand something about that code.

Reply:
Ohhh wait I didn't realize this file also has a loop. I see what you mean now.

Reply:
I think I pushed a fix.

But I don't know if it's good because it appears that this line is not covered by any test:

https://github.com/facebook/react/blob/790b5246f691adafbf4b6a4b3fe2e6cc1370c43e/packages/react-reconciler/src/ReactFiberBeginWork.new.js#L2763

I tried writing the one you suggested (make SuspenseList child items memo'd) but that also passed with this line commented out. I'm not sure how to make it reproduce.

Reply:
Oh ok I found a way to write a test. nvm.

Reply:
https://github.com/facebook/react/pull/23095/commits/1fed15d4fe84994f7fd2f1a7b66ba674819ce791